azure-pipelines-ui-tests.dfe.yml

```diff
@@ -0,0 +1,212 @@
+      workingDirectory: tests/robot-tests
+  - task: PublishTestResults@2
+    displayName: Publish Test Results
+    inputs:
+      testResultsFiles: tests/robot-tests/test-results/xunit.xml
+      failTaskOnFailedTests: true
+  - task: PublishPipelineArtifact@1
+    displayName: Publish Pipeline Artifact
+    inputs:
+      path: tests/robot-tests/test-results/
+      artifactName: test-results-admin-and-public-2
+- job: Phase_1
+  displayName: Admin UI tests
+  timeoutInMinutes: 160
+  cancelTimeoutInMinutes: 10
+  condition: succeededOrFailed()
+  pool:
+    vmImage: ubuntu-20.04
+  steps:
+  - checkout: self
+    clean: true
+    fetchDepth: 5
+    fetchTags: false
+  - task: UsePythonVersion@0
+    displayName: Use Python 3.10
+    retryCountOnTaskFailure: 5
+    timeoutInMinutes: 5
+    inputs:
+      versionSpec: 3.10
+  - task: AzureKeyVault@2
+    displayName: 'Azure Key Vault: s101d01-kv-ees-01'
+    inputs:
+      ConnectedServiceName: 372f9261-ade1-429b-be3c-aebd417646ca
+      KeyVaultName: s101d01-kv-ees-01
+      SecretsFilter: ees-test-ADMIN-PASSWORD,ees-test-ANALYST-PASSWORD,ees-test-expiredinvite-password
+  - task: PythonScript@0
+    displayName: Admin UI tests
+    condition: succeededOrFailed()
+    inputs:
+      scriptPath: tests/robot-tests/scripts/run_tests_pipeline.py
+      arguments: --admin-pass  "$(ees-test-ADMIN-PASSWORD)" --analyst-pass  "$(ees-test-ANALYST-PASSWORD)" --expiredinvite-pass "$(ees-test-expiredinvite-password)" --env "dev" --file "tests/admin" --processes 2
+      workingDirectory: tests/robot-tests
+  - task: PythonScript@0
+    displayName: Admin UI tests - rerun failed suites
+    condition: not(succeeded())
+    inputs:
+      scriptPath: tests/robot-tests/scripts/run_tests_pipeline.py
+      arguments: --admin-pass  "$(ees-test-ADMIN-PASSWORD)" --analyst-pass  "$(ees-test-ANALYST-PASSWORD)" --expiredinvite-pass "$(ees-test-expiredinvite-password)" --env "dev" --file "tests/admin" --processes 2 --rerun-failed-suites
+      workingDirectory: tests/robot-tests
+  - task: PublishTestResults@2
+    displayName: Publish Test Results
+    condition: succeededOrFailed()
+    inputs:
+      testResultsFiles: tests/robot-tests/test-results/xunit.xml
+      failTaskOnFailedTests: true
+  - task: PublishPipelineArtifact@1
+    displayName: Publish Test Pipeline Artifact
+    condition: succeededOrFailed()
+    inputs:
+      path: tests/robot-tests/test-results/
+      artifactName: test-results-admin
+- job: Job_1
+  displayName: Admin & public suites
+  timeoutInMinutes: 160
+  cancelTimeoutInMinutes: 10
+  condition: succeededOrFailed()
+  pool:
+    vmImage: ubuntu-20.04
+  steps:
+  - checkout: self
+    clean: true
+    fetchDepth: 5
+    fetchTags: false
+  - task: UsePythonVersion@0
+    displayName: Use Python 3.10
+    retryCountOnTaskFailure: 5
```

Comment:
We should probably set this to 2 to avoid re-trying too much (i.e. if an environment has issues, we don't want to continually retry this task).